elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,410 @@
+            String fieldName = field.getName();
+            Class<?> parentType = parentProjection.getType();
+
+            // this field would either be a relationship field or an attribute field
+            if (entityDictionary.getRelationshipType(parentType, fieldName) != RelationshipType.NONE) {
+                // handle the case of a relationship field
+                final Class<?> relationshipType =
+                        entityDictionary.getParameterizedType(parentType, fieldName) == null
+                                ? entityDictionary.getType(parentType, fieldName)
+                                : entityDictionary.getParameterizedType(parentType, fieldName);
+
+                // build new entity projection with only entity type and entity dictionary
+                EntityProjection relationshipProjection = createProjection(field);
+
+                // add this relationship projection to its parent projection
+                parentProjection.getRelationships().put(fieldName, relationshipProjection);
+
+                return;
+            }
+
+            Class<?> attributeType = entityDictionary.getType(parentType, fieldName);
+            if (attributeType != null) {
+                Set<Attribute> attributes = parentProjection.getAttributes();
+                Attribute matched = parentProjection.getAttributeByName(fieldName);
+
+                if (matched == null) {
+                    // this is a new attribute, add it
+                    attributes.add(Attribute.builder().type(attributeType).name(fieldName).build());
+
+                    // update mutated attributes
+                    parentProjection.setAttributes(attributes);
+                }
+            } else {
+                throw new InvalidEntityBodyException(
+                        String.format("Unknown attribute field {%s.%s}.", parentProjection.getName(), fieldName));
+            }
+        } else {
+            throw new InvalidEntityBodyException(
+                    String.format("Unsupported selection type {%s}.", fieldSelection.getClass()));
+        }
+    }
+
+    /**
+     * Construct Elide {@link Pagination}, {@link Sorting}, {@link Attribute} from GraphQL {@link Argument} and
+     * add it to the {@link EntityProjection}.
+     *
+     * @param argument graphQL argument
+     * @param entityProjection projection that has this argument
+     */
+    private void addArgument(Argument argument, EntityProjection entityProjection) {
+        String argumentName = argument.getName();
+        Value argumentValue = argument.getValue();
+
+        if (isPaginationArgument(argumentName)) {
+            addPagination(argumentName, argumentValue, entityProjection);
+        } else if (isSortingArgument(argumentName)) {
+            addSorting(argumentValue, entityProjection);
+        } else {
+            Class<?> entityType = entityProjection.getType();
+
+            if (!entityDictionary.isValidField(entityType, argumentName)) {
```

Comment:
Arguments come in two flavors:
1. Arguments elide defines for filters, pagination, and sorting
2. attribute arguments which are up to the datastore to interpret.

The latter are not attribute names, but something like:
`{ playerStats { score(aggregation : sum) date(grain : day) } }`